### PR TITLE
fix: support for graphql-ruby v1.13

### PR DESCRIPTION
**Pull request by 1060ki**
- Added support for GraphQL-ruby v1.13.

The following error occurs since graphql-ruby v1.13.1.

```
  1) ApolloFederation::EntitiesField with the interpreter runtime behaves like entities field when a type with the key directive exists when a Query object is provided sets the Query as the owner to the _entities field
     Failure/Error:
       def to_graphql
         field_defn = super # Returns a GraphQL::Field
         field_defn.metadata[:federation_directives] = @federation_directives
         field_defn
       end

     ArgumentError:
       wrong number of arguments (given 1, expected 0)
     Shared Example Group: "entities field" called from ./spec/apollo-federation/entities_field_spec.rb:349
     # ./lib/apollo-federation/has_directives.rb:12:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/object.rb:137:in `block in to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/object.rb:136:in `each'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/object.rb:136:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/member/accepts_definition.rb:129:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/member/cached_graphql_definition.rb:52:in `to_graphql'
     # ./lib/apollo-federation/has_directives.rb:13:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/member/cached_graphql_definition.rb:28:in `deprecated_to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/member/cached_graphql_definition.rb:21:in `graphql_definition'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema.rb:947:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/member/accepts_definition.rb:129:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema/member/cached_graphql_definition.rb:52:in `to_graphql'
     # ./.bundle/ruby/2.7.0/gems/graphql-1.13.1/lib/graphql/schema.rb:910:in `graphql_definition'
     # ./spec/apollo-federation/entities_field_spec.rb:74:in `block (5 levels) in <top (required)>'
```

This is because `to_graphql` is passed parameter argument.
ref: https://github.com/rmosolgo/graphql-ruby/commit/f449669e54601a1ad213c97767c810df83a92414
So I also fixed `to_graphql` of ApolloFederation::HasDirectives so that it can accept any parameter argument.

### CODE REVIEW

**Comment by Austio** [comment]
@1060ki Thanks so much for the PR!  Could you add more details in the description for why the changes are needed here (also if you could what changed in graphql-ruby to require)

**Comment by 1060ki** [comment]
@Austio Thanks for the reply.I have added the description.
Best regards.

**Comment by daemonsy** [comment]
Hi @1060ki thanks for the PR to upgrade this to 1.13 and taking a while to respond.

I have a [PR](https://github.com/Gusto/apollo-federation-ruby/pull/160) that works on bringing this library up to `1.13.x`, but I'm inclined to take yours for the upgrade if you're still keen on working on it. 

That way, I can focus more on doing yak shaving, removing the blockers that make it harder to ship the upgrade. 

I'll leave a few notes in the code review.

File: lib/apollo-federation/has_directives.rb [hunk 1]
```diff
@@ -9,8 +9,8 @@ def add_directive(name:, arguments: nil)
       @federation_directives << { name: name, arguments: arguments }
     end
 
-    def to_graphql
-      field_defn = super # Returns a GraphQL::Field
+    def to_graphql(**args)
```
Comment (daemonsy) [inline lib/apollo-federation/has_directives.rb:12]: Thanks for linking to [this](https://github.com/rmosolgo/graphql-ruby/commit/f449669e54601a1ad213c97767c810df83a92414). I know the tests were failing but that was in the 1.9.x test suite We [removed support](https://github.com/Gusto/apollo-federation-ruby/pull/162/files) for 1.9.x and now tests can pass without adding args. 

But the key question IMO is what are the args for? From reading the commit It seems like args were actually added to support passing `silence_deprecation_warnings: true`  to `to_graphql`. 

1. Should we be using `to_graphql(**args, **kwargs)`? This captures both positional and keyword args to the function?
2. With our library patching `to_graphql`, does the silencing of args still work as expected?

**Review by daemonsy — COMMENTED** [review]
Some other considerations 

- [ ] We need to run the appraisal suite for GraphQL 1.13. See  [this change](https://github.com/Gusto/apollo-federation-ruby/pull/160/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R105) 
- [ ] I rewrote this [messy spec](https://github.com/Gusto/apollo-federation-ruby/pull/160/files#diff-c1d0d606872073382ccb9cdd802cdadab8e0185f3841530375860a18b86ab3e9R437) that should fail if you run the 1.13.x test suite. I can ship this to main. That's a todo for myself.

**Comment by daemonsy** [comment]
Hi @1060ki , I'd like to get the GraphQL 1.13 support out earlier, will be merging your commit into my branch. Happy Friday!

**Pull request merged**